File: backend/src/main/java/net/zorphy/backend/site/jolly/dto/game/GameStats.java
package net.zorphy.backend.site.jolly.dto.game;

import net.zorphy.backend.main.dto.game.stats.GameSpecificStats;
import net.zorphy.backend.main.dto.game.stats.metrics.GameStatsMetrics;

import java.time.Duration;


public record GameStats(
        int roundsPlayed,
        double roundWinRate,
        GameStatsMetrics<Double> roundScoreMetrics,
        GameStatsMetrics<Duration> roundDurationMetrics,
        double outInOneRate,
        double closedRate
) implements GameSpecificStats {
}

File: frontend/src/app/sites/jolly/dto/game/GameStats.ts
import {LinkedGameStats} from "../../../../main/dto/games/stats/LinkedGameStats";

export interface GameStats {
    roundsPlayed: number,
    roundWinRate: number,
    minRoundScore: LinkedGameStats<number>,
    avgRoundScore: number,
    maxRoundScore: LinkedGameStats<number>,
    minRoundDuration: LinkedGameStats<string>,
    avgRoundDuration: string,
    maxRoundDuration: LinkedGameStats<string>,
    outInOneRate: number,
    closedRate: number
}